src/components/forms/NumberField/NumberField.stories.mdx

```diff
@@ -35,22 +43,118 @@ const generalTest = async canvasElement => {
   }}
 />
 
+# Number inputs
```

Comment:
I've tried to summarize our back-and-forth thoughts/discussions in some proper documentation with recommendations on when to use what.